### ISSUE TITLE
Switch workspace broken in gnome 3.30

### ISSUE THREAD

**Issue by treba123** [mpiannucci/gnome-shell-extended-gestures#47]
I'm currently on git master for mutter/gnome-shell (3.29.91+)

"Toggle Overview" and "Show App Drawer" and "Cycle Applications" still work fine, but the same gesture with "Switch Workspace" fails.

journalctl output:
```
Aug 29 00:39:47 robert.thinkpad-x230 gnome-shell[2081]: JS ERROR: Exception in callback for signal: activated: TypeError: this._switchData is null
                                                        _actionSwitchWorkspace@resource:///org/gnome/shell/ui/windowManager.js:1094:13
                                                        wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                                        TouchpadGestureAction<._doAction@/home/robert/.local/share/gnome-shell/extensions/extendedgestures@mpiannucci.github.com/extension.js:143:17
                                                        wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                                        _emit@resource:///org/gnome/gjs/modules/signals.js:128:27
                                                        TouchpadGestureAction<._checkActivated@/home/robert/.local/share/gnome-shell/extensions/extendedgestures@mpiannucci.github.com/extension.js:73:9
                                                        wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                                        TouchpadGestureAction<._handleEvent@/home/robert/.local/share/gnome-shell/extensions/extendedgestures@mpiannucci.github.com/extension.js:90:17
                                                        wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
```

**Comment by mpiannucci** [comment]
Thanks for the report. I have only tested up to 3.28 so this is good to track. I will have to look through the source to see what changed on the gnome side. 

Just curious, does the built in four finger swipe to change workspace still work as expected?

**Comment by mpiannucci** [comment]
Where it crashed:

![image](https://user-images.githubusercontent.com/1832671/44798511-aad3be80-ab7f-11e8-999e-8fd9112d614c.png)

https://github.com/GNOME/gnome-shell/blob/6b5117677dd0abf7b2641b5673cfefd87622f2a2/js/ui/windowManager.js#L1095

**Comment by treba123** [comment]
Hej, thanks for the replies :)
Just tested it and switching the desktop with four fingers still works perfectly fine

**Comment by mpiannucci** [comment]
Fixed with #50 . 

There is more work to do with better incremental switching but for now this can be closed
